### PR TITLE
CODE RUB: Upgrade to Standard `2.10.0`

### LINKED ISSUE #20: FOUNDATIONS: Upgrade Test Exceptions v2.10.0
A short time back there was an upgrade to the Standard version v2.10.0

Here is the video link:
https://youtu.be/oyJvszELGwE?si=-bD6BfdTm_aecO3_

It will be far more appropriate to refactor the test base now, before it may become a larger number to do. It is a tedious process not a hard one.

### PR DESCRIPTION
**Pull request by pratikkaje**
Closes #20

### CODE REVIEW

File: ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs [hunk 1]
```diff
@@ -15,9 +15,11 @@ public class FailedTransactionStorageException : Xeption
...
             : base(message: "Failed transaction storage error occurred, contact support.", innerException)
-        {
+        { }
 
-        }
+        public FailedTransactionStorageException(string message, Exception innerException)
+            : base(message: "Failed transaction storage error occurred, contact support.", innerException)
+        { }
 
         public FailedTransactionStorageException(SqlException sqlException)
```
Comment (glhays) [inline ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: Remove this constructor it does not belong here.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/NotFoundTransactionException.cs [hunk 2]
```diff
@@ -13,5 +13,9 @@ public class NotFoundTransactionException : Xeption
         public NotFoundTransactionException(Guid transactionId)
             : base(message: $"Couldn't find transaction with Id {transactionId}.")
```
Comment (glhays) [inline ExpenseTracker.Core/Models/Transactions/Exceptions/NotFoundTransactionException.cs]:
```suggestion
            : base(message: $"Transaction not found with Id {transactionId}.")
```

Comment (glhays) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/NotFoundTransactionException.cs]: or something other than Couldn't, just needs to be something more business appropriate.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs [hunk 3]
```diff
@@ -11,5 +11,8 @@ public class NullTransactionException : Xeption
     {
         public NullTransactionException() : base(message: "Transaction is null.")
```
Comment (glhays) [inline ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs]:
```suggestion
        public NullTransactionException()
            : base(message: "Transaction is null.")
```

File: ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs [hunk 4]
```diff
@@ -11,5 +11,8 @@ public class NullTransactionException : Xeption
...
         { }
+
+        public NullTransactionException(string message) : base(message: message)
```
Comment (glhays) [inline ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs]:
```suggestion
        public NullTransactionException(string message)
            : base(message: message)
```

Comment (glhays) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs]: General rule of good coding is to be consistent in your classes, if you break them in some do it for all.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/AlreadyExistsTransactionException.cs [hunk 5]
```diff
@@ -13,5 +13,9 @@ public class AlreadyExistsTransactionException : Xeption
         public AlreadyExistsTransactionException(Exception innerException)
             : base(message: "Transaction with the same id already exists.", innerException)
```
Comment (glhays) [inline ExpenseTracker.Core/Models/Transactions/Exceptions/AlreadyExistsTransactionException.cs:14]: General rule of good coding is if your going to use the parameter hints you should for all. I know that here with the innerException it looks redundent.

```suggestion
            : base(
                message: "Transaction with the same id already exists.",
                innerException: innerException)
```

Comment (glhays) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/AlreadyExistsTransactionException.cs:14]: do the same for all the rest.

File: ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs [hunk 6]
```diff
@@ -29,7 +29,9 @@ public async Task ShouldThrowCriticalDependencyExceptionOnAddIfSqlErrorOccurrsAn
                 new FailedTransactionStorageException(sqlException);
 
             var expectedTransactionDependencyException =
-                new TransactionDependencyException(failedTransactionStorageException);
+                new TransactionDependencyException(
+                    message: "Transaction dependency error occurred, contact support."
+                    , innerException: failedTransactionStorageException);
```
Comment (glhays) [inline ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: move the comma up to the end of the previous line

Comment (glhays) [reply ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: do the same for the others

**Review by glhays — COMMENTED** [review]
@pratikkaje see review comments. I did not do a complete as many of the comments will be repeated.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs (hunk 1 above)
Comment (pratikkaje) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: @glhays I am using this constructor for DBUpdateException shown below. Shall I keep it or remove it?

image

File: ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs (hunk 3 above)
Comment (pratikkaje) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs]: Fixed.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs (hunk 4 above)
Comment (pratikkaje) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/NullTransactionException.cs]: Fixed.

File: ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs (hunk 6 above)
Comment (pratikkaje) [reply ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: Resolved.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/AlreadyExistsTransactionException.cs (hunk 5 above)
Comment (pratikkaje) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/AlreadyExistsTransactionException.cs:14]: Resolved.

File: ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs (hunk 1 above)
Comment (glhays) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: Remove it.

Let me revise with some better context.
In your test, how does that constructor have a purpose? From what I see, it has no purpose. Your passing a DbUpdate exception to the localized exception in the test. So I do not see the purpose of the constructor. The TryCatch will intercept the exception throw the appropriate categorization exception. In The-Standard it is you localize then categorize your exception. 
Does the test pass when it is removed?

Comment (pratikkaje) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: It seems the intention here is to verify the message during the localization process of the DbUpdateException. If a developer decides to alter the exception message, the test will detect this change and result in failure. I also referred intertrack project and they also have the constructor as shown below.

image

Comment (glhays) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: @pratikkaje 
I do not see this constructor in theirs?????
```
public FailedTransactionStorageException(SqlException sqlException)
{
            this.sqlException = sqlException;
}
Quote reply

Comment (glhays) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: And yes your are correct about your "seems the intention is"

Comment (pratikkaje) [reply ExpenseTracker.Core/Models/Transactions/Exceptions/FailedTransactionStorageException.cs]: @glhays Thank you for your feedback. I believe I now grasp the issue. I have proceeded to remove the constructor mentioned below. Please correct me if I am still missing the point. 😅

public FailedTransactionStorageException(SqlException sqlException)
{
            this.sqlException = sqlException;
}

File: ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs [hunk 7]
```diff
@@ -26,10 +26,16 @@ public async Task ShouldThrowCriticalDependencyExceptionOnAddIfSqlErrorOccurrsAn
             SqlException sqlException = GetSqlException();
 
             var failedTransactionStorageException =
-                new FailedTransactionStorageException(sqlException);
+                new FailedTransactionStorageException(
+                    message: "Failed transaction storage error occurred, contact support.",
+                    innerException: sqlException
+                    );
```
Comment (glhays) [inline ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: Move these all up to the end of previouc line.
There are many of these do it for all of them.

Comment (glhays) [reply ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: @pratikkaje this occurs on many of the tests, maybe a copy paste operation result.
Fix these and I will take another review of the PR.

Comment (pratikkaje) [reply ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: @glhays I'm sorry, I'm not quite following the comment. Could you please clarify?

Comment (glhays) [reply ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: Just move the ); on line 32 up to the end of 31. And all the others.

Comment (pratikkaje) [reply ExpenseTracker.Core.Tests.Unit/Services/Foundations/Transactions/TransactionServiceTests.Exceptions.Add.cs]: Fixed. 👍

File: ExpenseTracker.Core/Services/Foundations/Transactions/TransactionService.Validations.cs
Comment (glhays) [inline ExpenseTracker.Core/Services/Foundations/Transactions/TransactionService.Validations.cs:1]: Fix the indent formatting lines 119 -> 133

**Review by glhays — APPROVED** [review]

**Review by glhays — APPROVED** [review]
Fix the above, Looks good to me

File: ExpenseTracker.Core/Services/Foundations/Transactions/TransactionService.Validations.cs
Comment (pratikkaje) [reply ExpenseTracker.Core/Services/Foundations/Transactions/TransactionService.Validations.cs:1]: Fixed

**Pull request merged**